### PR TITLE
Board- Attack, Board- Placeship

### PR DESCRIPTION
**Pull request by isvpursuit**
Okay, sorry this is a lot, but now all of the functions should be complete.

Added a bunch more testing, everything is checking out, not sure why the front end isn't reading it. I looked through game.js but the only line I could see that appeared to be looking for ships was on line 43,
``game.playersBoard.ships.forEach((ship) => ship.occupiedSquares.forEach((square) => {``
but I can't see how it's reading what 'ships' is.

### CODE REVIEW

**Comment by cookema** [comment]
I was able to load your branch into IntelliJ and run the board test. Those all passed like you mentioned. Just looking over the Board code now.

File: src/main/java/cs361/battleships/models/Board.java [hunk 1]
```diff
@@ -5,44 +5,138 @@
...
 	 */
 	public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
-		// TODO Implement
-		return false;
+
+		// set ship location
+		ship.setOccupiedSquares(x, y, isVertical);
+
+		// validate valid ship type
+		if (ship.getShipType() != "MINESWEEPER" && ship.getShipType() != "DESTROYER" && ship.getShipType() != "BATTLESHIP") {
```
[the remark this answers is not in the document]
Comment (caiusb) [reply src/main/java/cs361/battleships/models/Board.java]: Also, this check should be done in `Ship`.

[the remark this answers is not in the document]
Comment (willijo8) [reply src/main/java/cs361/battleships/models/Board.java]: I am working on this change

File: src/main/java/cs361/battleships/models/Board.java [hunk 2]
```diff
@@ -5,44 +5,138 @@
...
+			return false;
+		}
+
+		// Need to generate occupied squares for this to work properly.
+		List<Square> shipPlacement = ship.getOccupiedSquares();
+
+		//Bounds validation on ship
```
[the remark this answers is not in the document]
Comment (willijo8) [reply src/main/java/cs361/battleships/models/Board.java]: I am working on this change

File: src/main/java/cs361/battleships/models/Board.java [hunk 3]
```diff
@@ -5,44 +5,138 @@
...
+			}
+		}
+
+		//Validated that result is not INVALID
+		this.attacks.add(atackResult);
+
+		//Check for ship at attack location
+		// ***this currently only checks for hit and not sunk or surrender
+		for (Ship ship: this.ships) {
+			List<Square> occupied = ship.getOccupiedSquares();
```
[the remark this answers is not in the document]
Comment (willijo8) [reply src/main/java/cs361/battleships/models/Board.java]: I am working on this change

**Pull request merged**